test/module_dir_import_test.py

```diff
@@ -0,0 +1,9 @@
+import unittest
+
+
+class TestModuleDirImport(unittest.TestCase):
```

Comment:
Weirdly we didn't seem to have anything that tested this sufficiently directly - I would have thought we must have but I got `ModuleDirImport` through them all with a deliberately busted implementation of `exec_module`...